### PR TITLE
Genome2

### PR DESCRIPTION
**Pull request by Brandon123b**
Genus/species name implemented

-- Old nematodes have new random names
-- Names change as new generations of nematodes change

### CODE REVIEW

File: NeuralNetwork/NeatNN.js [hunk 1]
```diff
@@ -148,51 +150,60 @@ class NeatNN {
...
         var mutateAddConnectionChance =     0.05 * NeatNN.MUTATION_MULTIPLIER;
         var mutateRemoveConnectionChance =  0.05 * NeatNN.MUTATION_MULTIPLIER;
         var mutateAddNodeChance =           0.03 * NeatNN.MUTATION_MULTIPLIER;
-        var mutateRemoveConnectionChance =  0.03 * NeatNN.MUTATION_MULTIPLIER;
+        var mutateRemoveNodeChance =        0.03 * NeatNN.MUTATION_MULTIPLIER;
 
         // Mutate the weight of a random connection
         while (mutateWeightChance > 0 && Math.random() < mutateWeightChance){
             mutateWeightChance -= 1;
+            this.lastDiff += .2;
```
Comment (NicolasWinsten) [inline NeuralNetwork/NeatNN.js]: for the separation increments, they should be stored as static constants in nematode.js like
Nemtaode.WEIGHT_CHANGE_SEP = 0.2

If this feels like we don't want Nematode logic inside NeatNN, I agree.  So maybe mutation calls should be confined to nematode.js (all I mean is transferring this Mutate() code to Nematode.js)

Comment (NicolasWinsten) [reply NeuralNetwork/NeatNN.js]: this change would also remove the need for the extra data+method of GetLastDiff() since we can calculate the diff in Nematode.js

Comment (Brandon123b) [reply NeuralNetwork/NeatNN.js]: I do really want there to be a single Mutate() function inside NeatNN because having to move the mutate() to Nematode.js would be moving NN code to nematode and would not be good. We could use statics in Nematode.js, but I really want to avoid coupling the NN with ANY non-NN classes. I want the NN to be able to run on its own without any dependencies. (Node and NNDisplay are part of NN) So if you want constants, we should put them in NeatNN. I figured that these were fine as non-statics because it may be unclear what they do when changing them, so I figured it may be fine to just have them there. I also didn't want to make 6 (Or maybe 3?) statics in the NN for a fairly minor feature unrelated to the NN. That's why I didn't have Mutate() return the diff and instead have a separate function to get it that would go unused if I moved the NN to another project. 

TLDR: Statics are fine if you want them, but I want them in NeatNN and not any unrelated classes to avoid unrelated coupling.

Comment (Brandon123b) [reply NeuralNetwork/NeatNN.js]: If you really want to complicate it with statics in Nematode.js, then we could have the NN store the exact change in terms of weights, biases, and node changes; then have GetLastDiff() take the static constants as parameters to have statics in nematodes without causing problems in the NN. Then GetLastDiff can compute the diff itself. (Or just have GetLastDiff return an array of changes or something and nematode.js can figure out what to do with it)

Comment (NicolasWinsten) [reply NeuralNetwork/NeatNN.js]: Right now, the NeatNN is coupled with Nematode logic, and I don't like that.  The mechanism for mutation (logic for adding connections / nodes) belongs in NeatNN, but how often and the probability of mutation belongs in Nematode.  So where it stands, NeatNN is coupled with Nematode logic, and I want to decouple it by moving the Nematode part of mutation out of NeatNN.

Comment (Brandon123b) [reply NeuralNetwork/NeatNN.js]: Do you want a way for the nematode to set the probabilities using a function like SetProbabilities(...)? I think it is better for the static probabilities to be in the NeatNN, but if you want to have the probabilities in the nematode as well, then we can make a function to pass them to the NN. I believe that these probabilities are less of a nematode specific thing and more of a generic probabilities. So I don't believe that these probabilities are directly nematode 
logic.
I do not believe the NN is coupled with the nematode except for the diff because the probabilities should be generic, not nematode specific. They may need to be modified per project, but that's not really coupling the logic, just different needs.

**Review by NicolasWinsten — APPROVED** [review]
